gradio/chat_interface.py

```diff
@@ -394,15 +396,34 @@ async def _submit_fn(
         self,
         message: str,
         history_with_input: list[list[str | None]],
+        request: gr.Request,
         *args,
     ) -> tuple[list[list[str | None]], list[list[str | None]]]:
         history = history_with_input[:-1]
+
+        inputs = [message, history, *args]
+
+        fn_signature = inspect.signature(self.fn)
+
+        request_param = next(
+            (
+                param_name
+                for param_name, param in fn_signature.parameters.items()
+                if param.annotation == gr.Request
+            ),
+            None,
+        )
+
+        if request_param:
```

Comment:
Is this check even necessary, I think `special_args` will take care of it either way

Reply:
You are right, removed the check. My mistake.